src/jit/gentree.h

```diff
@@ -4097,6 +4101,30 @@ inline bool GenTree::IsZero()
     return false;
 }
 
+// Note that, unlike IsZero(), this only returns true only for integral types.
+inline bool GenTree::IsIOne()
+{
+    if ((gtOper == GT_CNS_INT) && (gtIntConCommon.IconValue() == 1))
+        return true;
+
+    if ((gtOper == GT_CNS_LNG) && (gtIntConCommon.LngValue() == 1))
+        return true;
+
+    return false;
```

Comment:
Nit: this whole function could be reduced to `return ((gtOper == GT_CNS_INT) && (gtIntConCommon.IconValue() == 1) || ((gtOper == GT_CNS_LNG) && (gtIntConCommon.LngValue() == 1));` (similarly for `IsIMinusOne`). If you feel that the current style is more readable, that's fine--I don't have a particularly strong opinion here.